upstream-sha256: 1fa17eda4d1a1acfee7e936852c07ec40d786dc2ea68e259a56da2bede04aaec
actual-sha256: f4b2bad101ab11d69d13840329726e98ec57123a146cb0e09bc921ad7653bfb2
--- upstream.expected
+++ tolk.actual
@@ -19,33 +19,45 @@
     float4 val14 = (*((float4*)((data2_24+(alu0+8)))));
     float4 val15 = (*((float4*)((data2_24+alu0))));
     int alu1 = (Lidx1*3);
-    float alu2 = (val0?val15[0]:((float)(-__builtin_inff())));
-    float alu3 = (val1?val15[1]:((float)(-__builtin_inff())));
-    float alu4 = (val6?val13[0]:((float)(-__builtin_inff())));
-    float alu5 = (val7?val13[1]:((float)(-__builtin_inff())));
-    float alu6 = (val10?val14[0]:((float)(-__builtin_inff())));
-    float alu7 = (val11?val14[1]:((float)(-__builtin_inff())));
-    float alu8 = (val4?val15[2]:((float)(-__builtin_inff())));
-    float alu9 = ((alu2<alu3)?alu3:alu2);
-    float alu10 = (val8?val13[2]:((float)(-__builtin_inff())));
-    float alu11 = ((alu4<alu5)?alu5:alu4);
-    float alu12 = (val2?val14[2]:((float)(-__builtin_inff())));
-    float alu13 = ((alu6<alu7)?alu7:alu6);
-    float alu14 = (val5?val15[3]:((float)(-__builtin_inff())));
-    float alu15 = ((alu9<alu8)?alu8:alu9);
-    float alu16 = (val9?val13[3]:((float)(-__builtin_inff())));
-    float alu17 = ((alu11<alu10)?alu10:alu11);
-    float alu18 = (val3?val14[3]:((float)(-__builtin_inff())));
-    float alu19 = ((alu13<alu12)?alu12:alu13);
-    float alu20 = ((alu15<alu14)?alu14:alu15);
-    float alu21 = ((alu17<alu16)?alu16:alu17);
-    float alu22 = ((alu19<alu18)?alu18:alu19);
-    float alu23 = ((alu20<val12)?val12:alu20);
-    float alu24 = ((alu21<val12)?val12:alu21);
-    float alu25 = ((alu22<val12)?val12:alu22);
-    *(data0_6+(alu1+1)) = alu24;
-    *(data0_6+(alu1+2)) = alu25;
-    *(data0_6+alu1) = alu23;
+    float alu2 = (val10?val14[0]:((float)(-__builtin_inff())));
+    float alu3 = (val11?val14[1]:((float)(-__builtin_inff())));
+    float alu4 = (val2?val14[2]:((float)(-__builtin_inff())));
+    float alu5 = ((alu2!=alu2)?alu2:alu3);
+    float alu6 = ((alu3<alu2)?alu2:alu5);
+    float alu7 = (val3?val14[3]:((float)(-__builtin_inff())));
+    float alu8 = ((alu6!=alu6)?alu6:alu4);
+    float alu9 = ((alu4<alu6)?alu6:alu8);
+    float alu10 = ((alu9!=alu9)?alu9:alu7);
+    float alu11 = ((alu7<alu9)?alu9:alu10);
+    float alu12 = (val0?val15[0]:((float)(-__builtin_inff())));
+    float alu13 = (val1?val15[1]:((float)(-__builtin_inff())));
+    float alu14 = (val4?val15[2]:((float)(-__builtin_inff())));
+    float alu15 = ((alu12!=alu12)?alu12:alu13);
+    float alu16 = ((alu13<alu12)?alu12:alu15);
+    float alu17 = (val5?val15[3]:((float)(-__builtin_inff())));
+    float alu18 = ((alu16!=alu16)?alu16:alu14);
+    float alu19 = ((alu14<alu16)?alu16:alu18);
+    float alu20 = ((alu19!=alu19)?alu19:alu17);
+    float alu21 = ((alu17<alu19)?alu19:alu20);
+    float alu22 = (val6?val13[0]:((float)(-__builtin_inff())));
+    float alu23 = (val7?val13[1]:((float)(-__builtin_inff())));
+    float alu24 = (val8?val13[2]:((float)(-__builtin_inff())));
+    float alu25 = ((alu22!=alu22)?alu22:alu23);
+    float alu26 = ((alu23<alu22)?alu22:alu25);
+    float alu27 = (val9?val13[3]:((float)(-__builtin_inff())));
+    float alu28 = ((alu26!=alu26)?alu26:alu24);
+    float alu29 = ((alu24<alu26)?alu26:alu28);
+    float alu30 = ((alu29!=alu29)?alu29:alu27);
+    float alu31 = ((alu27<alu29)?alu29:alu30);
+    float alu32 = ((alu11!=alu11)?alu11:val12);
+    float alu33 = ((val12<alu11)?alu11:alu32);
+    float alu34 = ((alu21!=alu21)?alu21:val12);
+    float alu35 = ((val12<alu21)?alu21:alu34);
+    float alu36 = ((alu31!=alu31)?alu31:val12);
+    float alu37 = ((val12<alu31)?alu31:alu36);
+    *(data0_6+(alu1+1)) = alu37;
+    *(data0_6+(alu1+2)) = alu33;
+    *(data0_6+alu1) = alu35;
   }
 }
 ---
@@ -208,68 +220,68 @@
     int alu119 = (cast13>>1);
     int alu120 = (cast14>>1);
     float alu121 = (alu27-((float)(cast3)));
-    float alu122 = (alu34-((float)(cast4)));
-    float alu123 = (alu55-((float)(cast7)));
-    float alu124 = (((alu23<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu121)+0.0013392627006396651f)*alu121)+0.009618384763598442f)*alu121)+0.055503472685813904f)*alu121)+0.24022644758224487f)*alu121)+0.6931471824645996f)*alu121)+1.0f)*__builtin_bit_cast(float, (int)(((alu109+127)<<23)))*__builtin_bit_cast(float, (int)((((cast3-alu109)+127)<<23)))));
-    float alu125 = ((alu23<-150.0f)?0.0f:alu124);
-    float alu126 = (((alu30<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu122)+0.0013392627006396651f)*alu122)+0.009618384763598442f)*alu122)+0.055503472685813904f)*alu122)+0.24022644758224487f)*alu122)+0.6931471824645996f)*alu122)+1.0f)*__builtin_bit_cast(float, (int)(((alu110+127)<<23)))*__builtin_bit_cast(float, (int)((((cast4-alu110)+127)<<23)))));
-    float alu127 = ((alu30<-150.0f)?0.0f:alu126);
-    float alu128 = (alu24?((float)(__builtin_nanf(""))):alu125);
-    float alu129 = (alu31?((float)(__builtin_nanf(""))):alu127);
-    float alu130 = (((alu51<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu123)+0.0013392627006396651f)*alu123)+0.009618384763598442f)*alu123)+0.055503472685813904f)*alu123)+0.24022644758224487f)*alu123)+0.6931471824645996f)*alu123)+1.0f)*__builtin_bit_cast(float, (int)(((alu113+127)<<23)))*__builtin_bit_cast(float, (int)((((cast7-alu113)+127)<<23)))));
-    float alu131 = ((alu51<-150.0f)?0.0f:alu130);
-    float alu132 = (alu52?((float)(__builtin_nanf(""))):alu131);
-    float alu133 = (alu62-((float)(cast8)));
-    float alu134 = (((alu58<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu133)+0.0013392627006396651f)*alu133)+0.009618384763598442f)*alu133)+0.055503472685813904f)*alu133)+0.24022644758224487f)*alu133)+0.6931471824645996f)*alu133)+1.0f)*__builtin_bit_cast(float, (int)(((alu114+127)<<23)))*__builtin_bit_cast(float, (int)((((cast8-alu114)+127)<<23)))));
-    float alu135 = ((alu58<-150.0f)?0.0f:alu134);
-    float alu136 = (alu59?((float)(__builtin_nanf(""))):alu135);
-    float alu137 = (alu20-((float)(cast2)));
-    float alu138 = (((alu16<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu137)+0.0013392627006396651f)*alu137)+0.009618384763598442f)*alu137)+0.055503472685813904f)*alu137)+0.24022644758224487f)*alu137)+0.6931471824645996f)*alu137)+1.0f)*__builtin_bit_cast(float, (int)(((alu108+127)<<23)))*__builtin_bit_cast(float, (int)((((cast2-alu108)+127)<<23)))));
-    float alu139 = ((alu16<-150.0f)?0.0f:alu138);
-    float alu140 = (alu17?((float)(__builtin_nanf(""))):alu139);
-    float alu141 = (alu69-((float)(cast9)));
-    float alu142 = (alu76-((float)(cast10)));
-    float alu143 = (alu83-((float)(cast11)));
-    float alu144 = (((alu65<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu141)+0.0013392627006396651f)*alu141)+0.009618384763598442f)*alu141)+0.055503472685813904f)*alu141)+0.24022644758224487f)*alu141)+0.6931471824645996f)*alu141)+1.0f)*__builtin_bit_cast(float, (int)(((alu115+127)<<23)))*__builtin_bit_cast(float, (int)((((cast9-alu115)+127)<<23)))));
-    float alu145 = ((alu65<-150.0f)?0.0f:alu144);
-    float alu146 = (((alu72<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu142)+0.0013392627006396651f)*alu142)+0.009618384763598442f)*alu142)+0.055503472685813904f)*alu142)+0.24022644758224487f)*alu142)+0.6931471824645996f)*alu142)+1.0f)*__builtin_bit_cast(float, (int)(((alu116+127)<<23)))*__builtin_bit_cast(float, (int)((((cast10-alu116)+127)<<23)))));
-    float alu147 = ((alu72<-150.0f)?0.0f:alu146);
-    float alu148 = (alu66?((float)(__builtin_nanf(""))):alu145);
-    float alu149 = (alu73?((float)(__builtin_nanf(""))):alu147);
-    float alu150 = (((alu79<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu143)+0.0013392627006396651f)*alu143)+0.009618384763598442f)*alu143)+0.055503472685813904f)*alu143)+0.24022644758224487f)*alu143)+0.6931471824645996f)*alu143)+1.0f)*__builtin_bit_cast(float, (int)(((alu117+127)<<23)))*__builtin_bit_cast(float, (int)((((cast11-alu117)+127)<<23)))));
+    float alu122 = (((alu23<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu121)+0.0013392627006396651f)*alu121)+0.009618384763598442f)*alu121)+0.055503472685813904f)*alu121)+0.24022644758224487f)*alu121)+0.6931471824645996f)*alu121)+1.0f)*__builtin_bit_cast(float, (int)(((alu109+127)<<23)))*__builtin_bit_cast(float, (int)((((cast3-alu109)+127)<<23)))));
+    float alu123 = ((alu23<-150.0f)?0.0f:alu122);
+    float alu124 = (alu24?((float)(__builtin_nanf(""))):alu123);
+    float alu125 = (alu69-((float)(cast9)));
+    float alu126 = (((alu65<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu125)+0.0013392627006396651f)*alu125)+0.009618384763598442f)*alu125)+0.055503472685813904f)*alu125)+0.24022644758224487f)*alu125)+0.6931471824645996f)*alu125)+1.0f)*__builtin_bit_cast(float, (int)(((alu115+127)<<23)))*__builtin_bit_cast(float, (int)((((cast9-alu115)+127)<<23)))));
+    float alu127 = ((alu65<-150.0f)?0.0f:alu126);
+    float alu128 = (alu66?((float)(__builtin_nanf(""))):alu127);
+    float alu129 = (alu97-((float)(cast13)));
+    float alu130 = (((alu93<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu129)+0.0013392627006396651f)*alu129)+0.009618384763598442f)*alu129)+0.055503472685813904f)*alu129)+0.24022644758224487f)*alu129)+0.6931471824645996f)*alu129)+1.0f)*__builtin_bit_cast(float, (int)(((alu119+127)<<23)))*__builtin_bit_cast(float, (int)((((cast13-alu119)+127)<<23)))));
+    float alu131 = ((alu93<-150.0f)?0.0f:alu130);
+    float alu132 = (alu94?((float)(__builtin_nanf(""))):alu131);
+    float alu133 = (alu34-((float)(cast4)));
+    float alu134 = (((alu30<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu133)+0.0013392627006396651f)*alu133)+0.009618384763598442f)*alu133)+0.055503472685813904f)*alu133)+0.24022644758224487f)*alu133)+0.6931471824645996f)*alu133)+1.0f)*__builtin_bit_cast(float, (int)(((alu110+127)<<23)))*__builtin_bit_cast(float, (int)((((cast4-alu110)+127)<<23)))));
+    float alu135 = ((alu30<-150.0f)?0.0f:alu134);
+    float alu136 = (alu31?((float)(__builtin_nanf(""))):alu135);
+    float alu137 = (alu76-((float)(cast10)));
+    float alu138 = (((alu72<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu137)+0.0013392627006396651f)*alu137)+0.009618384763598442f)*alu137)+0.055503472685813904f)*alu137)+0.24022644758224487f)*alu137)+0.6931471824645996f)*alu137)+1.0f)*__builtin_bit_cast(float, (int)(((alu116+127)<<23)))*__builtin_bit_cast(float, (int)((((cast10-alu116)+127)<<23)))));
+    float alu139 = ((alu72<-150.0f)?0.0f:alu138);
+    float alu140 = (alu73?((float)(__builtin_nanf(""))):alu139);
+    float alu141 = (alu104-((float)(cast14)));
+    float alu142 = (((alu100<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu141)+0.0013392627006396651f)*alu141)+0.009618384763598442f)*alu141)+0.055503472685813904f)*alu141)+0.24022644758224487f)*alu141)+0.6931471824645996f)*alu141)+1.0f)*__builtin_bit_cast(float, (int)(((alu120+127)<<23)))*__builtin_bit_cast(float, (int)((((cast14-alu120)+127)<<23)))));
+    float alu143 = ((alu100<-150.0f)?0.0f:alu142);
+    float alu144 = (alu101?((float)(__builtin_nanf(""))):alu143);
+    float alu145 = (alu55-((float)(cast7)));
+    float alu146 = (((alu51<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu145)+0.0013392627006396651f)*alu145)+0.009618384763598442f)*alu145)+0.055503472685813904f)*alu145)+0.24022644758224487f)*alu145)+0.6931471824645996f)*alu145)+1.0f)*__builtin_bit_cast(float, (int)(((alu113+127)<<23)))*__builtin_bit_cast(float, (int)((((cast7-alu113)+127)<<23)))));
+    float alu147 = ((alu51<-150.0f)?0.0f:alu146);
+    float alu148 = (alu52?((float)(__builtin_nanf(""))):alu147);
+    float alu149 = (alu83-((float)(cast11)));
+    float alu150 = (((alu79<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu149)+0.0013392627006396651f)*alu149)+0.009618384763598442f)*alu149)+0.055503472685813904f)*alu149)+0.24022644758224487f)*alu149)+0.6931471824645996f)*alu149)+1.0f)*__builtin_bit_cast(float, (int)(((alu117+127)<<23)))*__builtin_bit_cast(float, (int)((((cast11-alu117)+127)<<23)))));
     float alu151 = ((alu79<-150.0f)?0.0f:alu150);
     float alu152 = (alu80?((float)(__builtin_nanf(""))):alu151);
-    float alu153 = (alu90-((float)(cast12)));
-    float alu154 = (((alu86<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu153)+0.0013392627006396651f)*alu153)+0.009618384763598442f)*alu153)+0.055503472685813904f)*alu153)+0.24022644758224487f)*alu153)+0.6931471824645996f)*alu153)+1.0f)*__builtin_bit_cast(float, (int)(((alu118+127)<<23)))*__builtin_bit_cast(float, (int)((((cast12-alu118)+127)<<23)))));
-    float alu155 = ((alu86<-150.0f)?0.0f:alu154);
-    float alu156 = (alu87?((float)(__builtin_nanf(""))):alu155);
-    float alu157 = (alu8-((float)(cast0)));
-    float alu158 = (((alu4<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu157)+0.0013392627006396651f)*alu157)+0.009618384763598442f)*alu157)+0.055503472685813904f)*alu157)+0.24022644758224487f)*alu157)+0.6931471824645996f)*alu157)+1.0f)*__builtin_bit_cast(float, (int)(((alu106+127)<<23)))*__builtin_bit_cast(float, (int)((((cast0-alu106)+127)<<23)))));
-    float alu159 = ((alu4<-150.0f)?0.0f:alu158);
-    float alu160 = (alu5?((float)(__builtin_nanf(""))):alu159);
-    float alu161 = (alu41-((float)(cast5)));
-    float alu162 = (alu97-((float)(cast13)));
-    float alu163 = (alu104-((float)(cast14)));
-    float alu164 = (((alu93<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu162)+0.0013392627006396651f)*alu162)+0.009618384763598442f)*alu162)+0.055503472685813904f)*alu162)+0.24022644758224487f)*alu162)+0.6931471824645996f)*alu162)+1.0f)*__builtin_bit_cast(float, (int)(((alu119+127)<<23)))*__builtin_bit_cast(float, (int)((((cast13-alu119)+127)<<23)))));
-    float alu165 = ((alu93<-150.0f)?0.0f:alu164);
-    float alu166 = (((alu100<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu163)+0.0013392627006396651f)*alu163)+0.009618384763598442f)*alu163)+0.055503472685813904f)*alu163)+0.24022644758224487f)*alu163)+0.6931471824645996f)*alu163)+1.0f)*__builtin_bit_cast(float, (int)(((alu120+127)<<23)))*__builtin_bit_cast(float, (int)((((cast14-alu120)+127)<<23)))));
-    float alu167 = ((alu100<-150.0f)?0.0f:alu166);
-    float alu168 = (alu94?((float)(__builtin_nanf(""))):alu165);
-    float alu169 = (alu101?((float)(__builtin_nanf(""))):alu167);
-    float alu170 = (((alu37<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu161)+0.0013392627006396651f)*alu161)+0.009618384763598442f)*alu161)+0.055503472685813904f)*alu161)+0.24022644758224487f)*alu161)+0.6931471824645996f)*alu161)+1.0f)*__builtin_bit_cast(float, (int)(((alu111+127)<<23)))*__builtin_bit_cast(float, (int)((((cast5-alu111)+127)<<23)))));
-    float alu171 = ((alu37<-150.0f)?0.0f:alu170);
-    float alu172 = (alu38?((float)(__builtin_nanf(""))):alu171);
-    float alu173 = (alu48-((float)(cast6)));
-    float alu174 = (((alu44<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu173)+0.0013392627006396651f)*alu173)+0.009618384763598442f)*alu173)+0.055503472685813904f)*alu173)+0.24022644758224487f)*alu173)+0.6931471824645996f)*alu173)+1.0f)*__builtin_bit_cast(float, (int)(((alu112+127)<<23)))*__builtin_bit_cast(float, (int)((((cast6-alu112)+127)<<23)))));
-    float alu175 = ((alu44<-150.0f)?0.0f:alu174);
-    float alu176 = (alu45?((float)(__builtin_nanf(""))):alu175);
+    float alu153 = (alu41-((float)(cast5)));
+    float alu154 = (((alu37<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu153)+0.0013392627006396651f)*alu153)+0.009618384763598442f)*alu153)+0.055503472685813904f)*alu153)+0.24022644758224487f)*alu153)+0.6931471824645996f)*alu153)+1.0f)*__builtin_bit_cast(float, (int)(((alu111+127)<<23)))*__builtin_bit_cast(float, (int)((((cast5-alu111)+127)<<23)))));
+    float alu155 = ((alu37<-150.0f)?0.0f:alu154);
+    float alu156 = (alu38?((float)(__builtin_nanf(""))):alu155);
+    float alu157 = (alu62-((float)(cast8)));
+    float alu158 = (((alu58<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu157)+0.0013392627006396651f)*alu157)+0.009618384763598442f)*alu157)+0.055503472685813904f)*alu157)+0.24022644758224487f)*alu157)+0.6931471824645996f)*alu157)+1.0f)*__builtin_bit_cast(float, (int)(((alu114+127)<<23)))*__builtin_bit_cast(float, (int)((((cast8-alu114)+127)<<23)))));
+    float alu159 = ((alu58<-150.0f)?0.0f:alu158);
+    float alu160 = (alu59?((float)(__builtin_nanf(""))):alu159);
+    float alu161 = (alu90-((float)(cast12)));
+    float alu162 = (((alu86<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu161)+0.0013392627006396651f)*alu161)+0.009618384763598442f)*alu161)+0.055503472685813904f)*alu161)+0.24022644758224487f)*alu161)+0.6931471824645996f)*alu161)+1.0f)*__builtin_bit_cast(float, (int)(((alu118+127)<<23)))*__builtin_bit_cast(float, (int)((((cast12-alu118)+127)<<23)))));
+    float alu163 = ((alu86<-150.0f)?0.0f:alu162);
+    float alu164 = (alu87?((float)(__builtin_nanf(""))):alu163);
+    float alu165 = (alu48-((float)(cast6)));
+    float alu166 = (((alu44<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu165)+0.0013392627006396651f)*alu165)+0.009618384763598442f)*alu165)+0.055503472685813904f)*alu165)+0.24022644758224487f)*alu165)+0.6931471824645996f)*alu165)+1.0f)*__builtin_bit_cast(float, (int)(((alu112+127)<<23)))*__builtin_bit_cast(float, (int)((((cast6-alu112)+127)<<23)))));
+    float alu167 = ((alu44<-150.0f)?0.0f:alu166);
+    float alu168 = (alu45?((float)(__builtin_nanf(""))):alu167);
+    float alu169 = (alu20-((float)(cast2)));
+    float alu170 = (((alu16<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu169)+0.0013392627006396651f)*alu169)+0.009618384763598442f)*alu169)+0.055503472685813904f)*alu169)+0.24022644758224487f)*alu169)+0.6931471824645996f)*alu169)+1.0f)*__builtin_bit_cast(float, (int)(((alu108+127)<<23)))*__builtin_bit_cast(float, (int)((((cast2-alu108)+127)<<23)))));
+    float alu171 = ((alu16<-150.0f)?0.0f:alu170);
+    float alu172 = (alu17?((float)(__builtin_nanf(""))):alu171);
+    float alu173 = (alu8-((float)(cast0)));
+    float alu174 = (((alu4<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu173)+0.0013392627006396651f)*alu173)+0.009618384763598442f)*alu173)+0.055503472685813904f)*alu173)+0.24022644758224487f)*alu173)+0.6931471824645996f)*alu173)+1.0f)*__builtin_bit_cast(float, (int)(((alu106+127)<<23)))*__builtin_bit_cast(float, (int)((((cast0-alu106)+127)<<23)))));
+    float alu175 = ((alu4<-150.0f)?0.0f:alu174);
+    float alu176 = (alu5?((float)(__builtin_nanf(""))):alu175);
     float alu177 = (alu14-((float)(cast1)));
     float alu178 = (((alu10<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu177)+0.0013392627006396651f)*alu177)+0.009618384763598442f)*alu177)+0.055503472685813904f)*alu177)+0.24022644758224487f)*alu177)+0.6931471824645996f)*alu177)+1.0f)*__builtin_bit_cast(float, (int)(((alu107+127)<<23)))*__builtin_bit_cast(float, (int)((((cast1-alu107)+127)<<23)))));
     float alu179 = ((alu10<-150.0f)?0.0f:alu178);
     float alu180 = (alu11?((float)(__builtin_nanf(""))):alu179);
-    *(data0_6+alu1) = (alu148+alu149+alu152+alu156+alu160);
-    *(data0_6+alu2) = (alu168+alu169+alu172+alu176+alu180);
-    *(data0_6+alu0) = (alu128+alu129+alu132+alu136+alu140);
+    *(data0_6+alu1) = (0.0f+alu128+alu140+alu152+alu164+alu176);
+    *(data0_6+alu2) = (0.0f+alu132+alu144+alu156+alu168+alu180);
+    *(data0_6+alu0) = (0.0f+alu124+alu136+alu148+alu160+alu172);
   }
 }
 ---
